src/pip/_internal/commands/download.py

```diff
@@ -133,7 +133,8 @@ def run(self, options, args):
             )
 
             downloaded = ' '.join([
-                req.name for req in requirement_set.successfully_downloaded
+                req.name for req in requirement_set.requirements.values()
+                if req.successfully_downloaded
```

Comment:
Q: does changing this to the following work?

```suggestion
                if not req.editable and not req.satisfied_by
```

Reply:
If it does... we could just drop the attribute basically.

Reply:
I'll give it a try. But even if it *does* work I think I'd want to hide it behind a `successfully_downloaded()` method, because that expression gives no hint that it's checking if the requirement was downloaded :-(

Then at least the poor soul who looks at that code in the future stands some chance of knowing what's going on!

Reply:
Hmm, the tests work with this change. But I did grep through the tests for "Successfully downloaded", and didn't get any hits, so I'm not actually sure we test for this.

So I'd prefer to stick with the current PR, as it's behaviourally the same as the existing code. We can add a test for the behaviour, and then refactor further, in a follow-up. For now, this decouples the download command from the `RequirementSet`, which is the important step at the moment.